### PR TITLE
Chore/update with supa

### PR DESCRIPTION
**Pull request by saltcod**
branched off @MildTomato's PR here: 
 https://github.com/vercel/next.js/pull/69308

- updates readme with shadcn/ui re-install instructions
- moves moves functions to actions.ts file 
- adds .env.example back

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 0ac6538bbc836c6fa1373a767937d46c197615ff

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/with-supabase/utils/cn.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-import { clsx, type ClassValue } from "clsx";
```
Comment (saltcod) [inline examples/with-supabase/utils/cn.ts:1]: dupe, already in utils.ts

File: examples/with-supabase/.env.example [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Update these with your Supabase details from your project settings > API
+# https://app.supabase.com/project/_/settings/api
```
Comment (saltcod) [inline examples/with-supabase/.env.example:2]: this was deleted from Jonny's branch

File: examples/with-supabase/README.md [hunk 3]
```diff
@@ -72,7 +73,9 @@ If you wish to just develop locally and not deploy to Vercel, [follow the steps
 
    Both `NEXT_PUBLIC_SUPABASE_URL` and `NEXT_PUBLIC_SUPABASE_ANON_KEY` can be found in [your Supabase project's API settings](https://app.supabase.com/project/_/settings/api)
 
-5. You can now run the Next.js local development server:
+5. This template comes with the default shadcn/ui style initialized. If you instead want the `New York` style, delete `components.json` and [re-install shadcn/ui](https://ui.shadcn.com/docs/installation/next)
```
Comment (saltcod) [inline examples/with-supabase/README.md:76]: since we initialize shadcn for them

File: examples/with-supabase/components/form-message.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ export function FormMessage({ message }: { message: Message }) {
         </div>
       )}
       {"error" in message && (
-        <div className="text-destructive-foreground border-l-2 border-destructive-foreground px-4">
+        <div className="text-red-500 border-l-2 border-destructive-foreground px-4">
```
Comment (saltcod) [inline examples/with-supabase/components/form-message.tsx:15]: colour wasn't working in light mode, so went with red

Comment (MildTomato) [reply examples/with-supabase/components/form-message.tsx:15]: weird, i'll check what's wrong.

shouddn't use hard coded colors

**Pull request closed without merging**